nexus/src/db/model.rs

```diff
@@ -1198,8 +1198,8 @@ impl From<params::VpcSubnetUpdate> for VpcSubnetUpdate {
             name: params.identity.name.map(Name),
             description: params.identity.description,
             time_modified: Utc::now(),
-            ipv4_block: params.ipv4_block.map(Ipv4Net),
-            ipv6_block: params.ipv6_block.map(Ipv6Net),
+            ipv4_block: Some(Ipv4Net(params.ipv4_block)),
+            ipv6_block: Some(Ipv6Net(params.ipv6_block)),
```

Comment:
is off here, though it may be that the former are correct and the latter are wrong.

It's a bit odd that `name` and `description` are optional on `IdentityMetadataUpdateParams` given that I thought we decided to go with the more strictly correct PUT scheme in which all fields are expected to be present in the request body. We probably just haven't updated all the code in accordance with that determination.

https://github.com/oxidecomputer/omicron/blob/939adec4de865caf4c0339fced7fdc75f8bfa5b0/common/src/api/external/mod.rs#L635-L640

If `name` and `description` are made optional in the params, I would expect the fields to remain optional on the diesel changeset because we can use that internally to make queries that only update some fields but not others. 

```rs
#[derive(AsChangeset)]
#[table_name = "vpc_subnet"]
pub struct VpcSubnetUpdate {
    pub name: Option<Name>,
    pub description: Option<String>,
    pub time_modified: DateTime<Utc>,
    pub ipv4_block: Option<Ipv4Net>,
    pub ipv6_block: Option<Ipv6Net>,
}
```

In that case the resolution to the above mismatch will be to wrap `name` and `description` in `Some`:

```rs
name: Some(Name(params.identity.name)),
description: Some(params.identity.description),
```

Reply:
Confirmed there's an issue for bringing our PUTs into line: #333